edifact-edi/2.3/modules/ROOT/pages/edifact-edi-connector-batching.adoc

```diff
@@ -0,0 +1,60 @@
+
+== Batching Stages
+
+Use the Anypoint EDIFACT EDI Connector Batch Write operation to merge individual EDIFACT messages into a single transmission to send to your trading partners.
+
+Implementing a batch involves two stages:
+
+. Accumulate the transactions.
+. Release the batch.
+
+=== Accumulate the Transactions
+
+To accumulate transactions for a batch:
+
+. Transform the payloads into EDIFACT messages using DataWeave and EDIFACT as you receive application messages (such as a JSON invoice) from the backend systems.
+. Create EDIFACT payloads using temporary control number keys in the EDIFACT Write operation. +
+This enables sending the accumulated EDIFACT as a batch later. The batch is created using control numbers that are not used in the final batch sent to the trading partner, for example, `<edifact:write doc:name="Write individual D97A ORDERS" doc:id="3d51215a-125d-461b-9791-0a939e5ebc45" config-ref="EDIFACT_Write_ORDERS" interchangeControlNumberKey="NTO-MYTHICAL-UNB-TMP"/>`.
+. Store the intermediate EDIFACT message payloads in your choice of storage, such as a database table:
+
+image::edifact-edi-connector-batch-flow-1.jpg[Accumulate the transactions]
+
+For example, this implementation of a Mule flow:
+
+* Receives JSON payloads from the backend systems.
+* Transforms to an `EDIFACT D97A ORDERS` message.
+* Generates the intermediate EDIFACT message payload using the EDIFACT Write operation.
+* Stores the EDIFACT payload in a database table.
+
+=== Release the Batch
+
+Trading partners have varying criteria for how they want transactions batched:
+
+* Release the batch at a specific time interval or a specific time of the day.
+* Release the batch when a specific number of transactions is accumulated.
+* Release the batch when the total size of the accumulated transactions is near a threshold.
+
+To release the batch:
+
+. Implement the logic or module to monitor the transactions accumulated in the temporary storage and determine when a particular batch is released.
+
+. Retrieve the accumulated intermediate EDIFACT messages from the temporary storage based on the batch criteria, group the transactions together, and then send the appended payload to the EDIFACT Batch Write operation. +
+You can either use the control number key on the Batch Write operation or leave the control number key fields empty at the time of Batch Write operation execution, allowing the EDIFACT EDI Connector to generate the control number sequence automatically based on the sender and receiver identifier combination:
+`<edifact:batch doc:name="Batch" doc:id="3ce80437-63e2-404c-b875-531708b7bfdb" config-ref="EDIFACT_Batch" interchangeControlNumberKey="NTO-MYTHICAL-UNB-TMP"> <edifact:batch-content ><![CDATA[#[vars.mergedInput]]]></edifact:batch-content> </edifact:batch>`.
```

Comment:
Here and at Line 20, I wonder if we should be formatting these strings as syntax examples, which sets them apart from the text for better visibility? Ask a teammate?